desc/objectives/utils.py

```diff
@@ -22,6 +32,13 @@ def factorize_linear_constraints(objective, constraint):  # noqa: C901
         Objective function to optimize.
     constraint : ObjectiveFunction
         Objective function of linear constraints to enforce.
+    things : Optimizable or tuple/list of Optimizable
```

Comment:
perhaps use `optimisables` as name for this parameters or something similar? `things` is not so descriptive :-)

Reply:
`things` is the argument name we use consistently throughout the rest of the code. It's not perfect, but `Optimizable` is already a class name so we couldn't think of anything better